bph_pal/bph_pal/bph_shell.py

```diff
@@ -0,0 +1,475 @@
+            self._write_pin_state("BP_RST", 0)
+            self._write_pin_state("EXT_V_EN", 0)
+        elif arg == "all":
+            print("external power on, USB power on, PHiLIP on")
+            subprocess.run("echo '1-1' |sudo tee /sys/bus/usb/"
+                           "drivers/usb/bind", shell=True)
+            self._pin_to_input("BP_RST")
+            self._write_pin_state("EXT_V_EN", 1)
+        elif arg == "usb":
+            print("external power off, USB power on, PHiLIP on")
+            subprocess.run("echo '1-1' |sudo tee /sys/bus/usb/"
+                           "drivers/usb/bind", shell=True)
+            self._pin_to_input("BP_RST")
+            self._write_pin_state("EXT_V_EN", 0)
+        else:
+            print("Invalid argument, no power change")
+
+    @staticmethod
+    def complete_power_source(text, line, begidx, endidx):
+        """Completes arg with possible power source settings"""
+        begidx = begidx
+        endidx = endidx
+        mline = line.partition(' ')[2]
+        offs = len(mline) - len(text)
+        power_opts = ['external', 'none', 'all', 'usb']
+        return [s[offs:] for s in power_opts if s.startswith(mline)]
+
+    def do_read_current(self, arg):
+        """Reads current to the dut thorugh the ext voltage line
+        Usage:
+            read_current [range]
+
+        Args:
+            range: The range settings
+                   [high_range<200mA, low_range<50uA]
+        """
+        value = (arg or '').split(' ')
+        func_args = [v for v in value if v]
+        try:
+            if len(func_args) is 0:
+                print("Current = {} A".format(self.curmon.get_current()))
+            elif len(func_args) is 1:
+                if func_args[0] == "high_range":
+                    self._write_pin_state("RES_BYP", 0)
+                elif func_args[0] == "low_range":
+                    self._write_pin_state("RES_BYP", 1)
+                else:
+                    raise KeyError
+                print("Current = {}A".format(self.curmon.get_current()))
+        except KeyError as exc:
+            print('Could not parse argument {}'.format(exc))
+        except (TypeError, ValueError, SyntaxError) as exc:
+            print(exc)
+
+    @staticmethod
+    def complete_read_current(text, line, begidx, endidx):
+        """Completes arg with possible ranges"""
+        begidx = begidx
+        endidx = endidx
+        mline = line.partition(' ')[2]
+        offs = len(mline) - len(text)
+        power_opts = ['high_range', 'low_range']
+        return [s[offs:] for s in power_opts if s.startswith(mline)]
+
+    def do_read_voltage(self, arg):
+        """Reads voltage to the dut thorugh the ext voltage line
+        Usage:
+            read_current [voltage_type=bus]
+
+        Args:
+            voltage_type: The voltage types...
+                          [bus, shunt]
+        """
+        value = (arg or '').split(' ')
```

Comment:
I think I already  mentioned that somewhere, when you use `split()` instead of `split(' ')` you can omit the next line and directly use `value` or assign that to `func_args`.

Reply:
or did you find a problem with that?